rust-toolchain.toml

```diff
@@ -0,0 +1,4 @@
+[toolchain]
+# Pin Rust 1.79 to avoid API breakage in `time` crate.
+channel = "1.79"
+components = [ "rustfmt", "rust-analyzer" ]
```

Comment:
Rather than pin old rust, maybe we should just add boxing where recommended. I'm sure that's happened in the upstream Hermes already.

Reply:
They did remove their `rust-toolchain.toml` file: https://github.com/informalsystems/hermes/pull/2706

I'm not sure what the `time` crate-related API breakage is, though

Reply:
Looks like they just set the `msrv` in the `Cargo.toml` files directly: https://github.com/informalsystems/hermes/blob/master/crates/relayer/Cargo.toml#L10

We have similar but on a different version: https://github.com/penumbra-zone/hermes/blob/main/crates/relayer/Cargo.toml#L5

Reply:
> I'm not sure what the `time` crate-related API breakage is, though

If you check out the main branch on this repo, without a `rust-toolchain.toml` file, and run `cargo check`, you'll see:

```
error[E0282]: type annotations needed for `Box<_>`
  --> /home/conor/.cargo/registry/src/index.crates.io-6f17d22bba15001f/time-0.3.34/src/format_description/parse/mod.rs:83:9
   |
83 |     let items = format_items
   |         ^^^^^
...
86 |     Ok(items.into())
   |              ---- type must be known at this point
   |
   = note: this is an inference error on crate `time` caused by an API change in Rust 1.80.0; update `time` to version `>=0.3.35` by calling `cargo
update`
```

Reply:
Hm I can `cargo check` on this branch without the `rust-toolchain.toml` file. This might be because of the Rust version I have installed?

```
$ rustc -V
rustc 1.82.0 (f6e511eec 2024-10-15)
```

Reply:
I pushed a commit to remove the `rust-toolchain.toml` and update the `msrv` to `1.79.0` everywhere